device.h

```diff
@@ -0,0 +1,24 @@
+#ifndef DEVICE_H
+#define DEVICE_H
+
+#include <string>
+#include <vector>
+#include "sensor.h"
+
+class Device {
+public:
+    Device(const std::string& type, const std::string& deviceId, bool status, const std::vector<Sensor> sensors);
+
+    const std::string getType() const;
+    const std::string getId() const;
+    bool getStatus() const;
+    const std::vector<Sensor> getSensors() const;
+
```

Comment:
Do you think you will also need setters for your classes or you don't expect to ever change the values set at initalization?(nothing wrong here, just curious)